pilz_trajectory_generation/include/pilz_trajectory_generation/robot_trajectory_helper.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2018 Pilz GmbH & Co. KG
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef ROBOT_TRAJECTORY_HELPER_H
+#define ROBOT_TRAJECTORY_HELPER_H
+
+#include <moveit/robot_trajectory/robot_trajectory.h>
+
+namespace pilz_trajectory_generation
+{
+
+class RobotTrajectoryHelper
+{
+public:
+    static void appendWithoutRedundantFirstState(robot_trajectory::RobotTrajectoryPtr &robot_trajectory,
```

Comment:
Due to the fact that the function does not manipulate the lifetime of both given trajectories, I suggest to use raw pointers. This is also suggest by the CppCoreGuidelines, see [here](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#r30-take-smart-pointers-as-parameters-only-to-explicitly-express-lifetime-semantics).

Reply:
done.